Switch existings tests to use equalsIgnoringHashCodes and add more tests.

jacob314 opened the pull request:
I'm open to keeping some of these tests on a local branch if we are concerned they will cause
too many busy work changes as widgets are updated. Just let me know which tests to keep out of the repo.
These tests were useful in refactoring the toStringDeep code but are perhaps a bit excessive
outside of that context.

jacob314 committed:
Switch existings tests to use equalsIgnoringHashCodes and add more tests.

Hixie commented on packages/flutter/test/rendering/paragraph_test.dart:
```diff
@@ -177,5 +176,29 @@ void main() {
+        ' ╘═╦══ text ═══\n'
+        '   ║ TextSpan:\n'
+        '   ║   "I polished up that handle so carefullee\n'
+        'That now I am the Ruler of the Queen\'s Navee!"\n'
```
can you add a TODO(ianh) to fix this?

Hixie commented:
LGTM. I'm fine with checking this in.

I think we would be better served by specific tests of each class, though.

jacob314 committed:
Add TODO(jacobr).

jacob314 committed:
Fix container test to work on all platforms.

Hixie commented on packages/flutter/test/widgets/animated_container_test.dart:
```diff
@@ -65,6 +65,41 @@ void main() {
+        ' │   color: Color(0xff0000ff)\n'
+        ' │ configuration: ImageConfiguration(bundle:\n'
+        ' │   PlatformAssetBundle#00000(), devicePixelRatio: 1.0, platform:\n'
+        ' │   macos)\n'
```
aren't you going to have the same issue here?

Hixie commented:
You can just force the platform to be whatever you want it to be, I think. Doesn't it come from the Theme?

jacob314 commented:
Fixed the issue that was causing the platform to not match the platform specified for tests in a separate CL.
Updated all tests to now expect android and verified the tests pass on macos.

Hixie commented on packages/flutter/test/widgets/overlay_test.dart:
```diff
@@ -25,5 +25,53 @@ void main() {
     expect(didBuild, isTrue);
+    final RenderObject theater = overlayKey.currentContext.findRenderObject();
+
+    // toStringDeep output is missing a trailing line break.
```
todo?

jacob314 replied:
done

Hixie commented:
![LGTM](http://junkyard.damowmow.com/577)

jacob314 committed:
Add TODO to overlay_test
